extension/jaegerremotesampling/extension.go

```diff
@@ -0,0 +1,87 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerremotesampling // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling"
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal"
+)
+
+var (
+	errAlreadyStarted = errors.New("the extension has been started already")
+	errNotYetStarted  = errors.New("the extension has not been started yet")
+)
+
+var _ component.Extension = (*jrsExtension)(nil)
+
+type jrsExtension struct {
+	started    bool
+	startedMu  sync.RWMutex
+	httpServer component.Component
+}
+
+func newExtension(cfg *Config, telemetry component.TelemetrySettings) (*jrsExtension, error) {
+	// TODO(jpkroehling): get a proper instance
+	var cfgMgr = internal.NewClientConfigManager()
+
+	httpServer, err := internal.NewHTTP(telemetry, cfg.HTTPServerSettings, cfgMgr)
+	if err != nil {
+		return nil, err
+	}
+
+	return &jrsExtension{
+		httpServer: httpServer,
+	}, nil
+}
+
+func (jrse *jrsExtension) Start(ctx context.Context, host component.Host) error {
+	jrse.startedMu.RLock()
+	if jrse.started {
+		return errAlreadyStarted
+	}
+	jrse.startedMu.RUnlock()
```

Comment:
We ensure that "Start/Stop" is called in this order and only once. There is a valid use-case "Start/Stop/Start/Stop" but not "Start/Start".

Reply:
Alright. I added it after I myself made a mistake calling the API from the tests, but if there's a guarantee that no Start/Start, or Shutdown (without Start) will be called, then we indeed don't need to track whether the component has started.